dom.go

```diff
@@ -911,14 +910,17 @@ func doCopy(c Component) Component {
 }
 
 // copyProps copies all struct fields from src to dst that are tagged with
-// `vecty:"prop"`.
+// `vecty:"prop"`, then replaces src with dst.
 //
-// If src and dst are different types, copyProps is no-op.
+// If src and dst are different types or non-pointers, copyProps panics.
 func copyProps(src, dst Component) {
 	s := reflect.ValueOf(src)
 	d := reflect.ValueOf(dst)
 	if s.Type() != d.Type() {
-		return
+		panic("vecty: internal error (attempted to copy properties of incompatible structs)")
```

Comment:
I changed this to panic, because if we get here, we've skipped a check somewhere, and this likely indicates a bug in vecty.